### ISSUE TITLE
Unable to launch firefox 48 with custom profile using Geckodriver

### ISSUE THREAD

**Issue by furqanmlk** [SeleniumHQ/selenium#2645]
## Meta -

OS:  Win 10 x64

Selenium Version:  
3.0
Browser:  
Firefox

Browser Version:  

Firefox 48.0.1
## Expected Behavior -

geckoriver should launch Firefox48 with custom profile
## Actual Behavior -

geckodriver still launches Firefox with new profile
## Steps to reproduce -

01: Zip default profile folder

02: Getting default profile location

string prof = (@"...\AppData\Roaming\Mozilla\Firefox\Profiles\6uqm9vnl.default.zip");

03: Converting to base64-encoded string
var prof1 = System.Text.Encoding.UTF8.GetBytes(prof);
var prof2 = System.Convert.ToBase64String(prof1);
FirefoxProfile profile = new FirefoxProfile(prof2);

04: Initializing FirefoxDriver
new FirefoxDriver(profile);

**Comment by MJBGO** [comment]
I can confirm the issue, I have the same problem and tried everything without any success ...

**Comment by everyx** [comment]
The same problem

**Comment by borginek** [comment]
Agree. It doesn't work :(

**Comment by SowBiT** [comment]
Same problem too.

**Comment by furqanmlk** [comment]
does anyone know how to **escalate** this issue ?

**Comment by Scr34mik** [comment]
I was able to do so with following code (sorry im using vb.net, ul lhave to convert it yourself)

        Dim FirefoxSrvc As FirefoxDriverService = FirefoxDriverService.CreateDefaultService()
        FirefoxSrvc.FirefoxBinaryPath = "C:\Program Files (x86)\Mozilla Firefox\firefox.exe"
        FirefoxSrvc.HideCommandPromptWindow = True

        Dim FirefoxOpts = New FirefoxOptions
        Dim FFLocalProfile As FirefoxProfile = New FirefoxProfile("C:\ff\joeluser")
        FirefoxOpts.Profile = FFLocalProfile

        driver = New FirefoxDriver(FirefoxSrvc, FirefoxOpts, New System.TimeSpan(0, 0, 60))

Hope this helps !

**Comment by furqanmlk** [comment]
Thanks Scr34mik for your help,

Unfortunately, I am getting error "corrupt deflate stream" at ,
driver = New FirefoxDriver(FirefoxSrvc, FirefoxOpts, New System.TimeSpan(0, 0, 60))

**Comment by dtext** [comment]
I can also confirm this issue. Looking for a long term solution to be able to use firefox for GUI tests.

**Comment by everyx** [comment]
I found that we can use https://github.com/mozilla/geckodriver/releases/tag/v0.7.1 then follow https://developer.mozilla.org/en-US/docs/Mozilla/QA/Marionette/WebDriver step by step.

```python
caps = DesiredCapabilities.FIREFOX
caps["marionette"] = True
caps["binary"] = settings.FIREFOX_PATH
driver = webdriver.Firefox(capabilities=caps)
```

It just work :)

**Comment by iKarthik** [comment]
I get a similar error, when I try to launch custom profile with Selenium
Environment
Firefox 48.0.2
Geckodriver v0.10

Has there been any updates regarding this?

**Comment by titusfortner** [comment]
Yes, Geckodriver v0.11 supports 'firefoxOptions' - https://github.com/mozilla/geckodriver/issues/228

**Comment by iKarthik** [comment]
Is there an expected release date for v0.11?

**Comment by furqanmlk** [comment]
Following code is working !
```
FirefoxDriverServise service = FirefoxDriverService.CreateDefaultService();
FirefoxProfile profile = new FirefoxProfile();
profile.AddExtension(Path.GetFullPath("path/to/extension.xpi"));
var options = new FirefoxOption();
options.Profile = profile;
var driver = new FirefoxDriver(service,options,TimeSpan.FromMinutes(30));
```

**Comment by cloudhx** [comment]
Hi furqanmlk, I tried your code but it still has the same error, I'm using the latest version for both selenium and geclodriver, could you please confirm this, thank you!

**Comment by redwards510** [comment]
I had a working setup with 0.11.1 and a custom profile (FFv49.0.2) for days. Then I did two things that screwed it up. I upgraded my Selenium.Support Nuget from 2.53.1 to v3 (to match my installed Selenium of v3.0) and installed Mozilla Nightly. I have no idea why either of those things would make it break, but it did. I started getting the "corrupt deflate stream" errror. 

I just got it to work again. I uninstalled Nightly. I then went back to my 49.0.2 mozilla and used ```firefox -P``` to create a new profile in a new directory. Set it to use that by default. Then I ran my code again, pointed to that new profile directory. It worked.

**Comment by couchpotatodude** [comment]
Try to remove 'user.js' file from profile directory.

**Comment by wcdeich4** [comment]
How can I use my real FireFox user profile?  (The existing profile that is used when I open FireFox every day.) 

The examples @furqanmlk and @Scr34mik provided are actually creating a new profile.  Maybe you can reuse the created profile - but it not my real user profile. 

Using the real user profile is highly advantageous b/c  sometimes there are firewall popups & security questions.  My real profile has already answered those questions & won't be challenged again.  A profile created by Selenium will get blocked by firewall popups in the middle of the automation test.

**Comment by asrafiq** [comment]
Older version of Firefox picked up the existing profile but now it creates a new profile for each test. can this be fixed please? The works on Chrome and IE like it did on Firefox before.

**Comment by titusfortner** [comment]
Geckodriver allows you to pass in the Base64 encoded zip of a profile directory (including your default) - https://github.com/mozilla/geckodriver#firefox-capabilities

The specifics on how to generate the correct Options parameter will vary based on which language bindings you are using.

**Comment by barancev** [comment]
Let's suppose this issue was fixed in geckodriver 0.11 (current version is 0.18)
Feel free to reopen it if it's not true.
